### ISSUE TITLE
Noticing some odd matches

### ISSUE THREAD

**Issue by jdalrymple** [sindresorhus/email-regex#9]
I was testing a few blobs of text with this regex rule but found some odd matches such as:

```
f="nr@context",c=e("gos")
```

This should return zero results, but, it returns : `f="nr@context",c=e("go` as a match. 

Have you seen some of the regex suggestions [here](https://stackoverflow.com/questions/42407785/regex-extract-email-from-strings)?

**Comment by jdalrymple** [comment]
I've been using:

```
/(?!\S*\.(?:jpg|png|gif|bmp|jpeg|css|js)(?:[\s\n\r]|$))[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]{3,65}\.[A-Za-z_-]{2,4}/gi
```

personally and its been pretty solid

**Comment by sindresorhus** [comment]
This regex started out for use with email validation. That's why it's so loose. For the use it's better to have false-positives, than false-negatives. I think we could make the regex stricter when the `exact: true` option is not passed.

**Comment by jdalrymple** [comment]
Fair enough

**Comment by niftylettuce** [comment]
I have published <https://github.com/niftylettuce/email-regex-safe> as an alternative to this package, with a lot of configurable options.  PR's welcome.

**Comment by babaralishah** [comment]
Hi 
I am on Angular

And using the below regex:

`'^[a-z0-9]+([._-]?[a-z0-9]+)+@[a-z0-9]+([._-]?[a-z0-9]+)+\\.[a-z]{2,3}$'`

But its not fulfilling my requirement, my requirement is below:

`saa5@me-d_d_u.co`

please anyone tell me, why this regex is not working for my above mentioned case.

Regards
Babar Ali Shah

**Comment by Richienb** [comment]
@sindresorhus Should `foo@bar` count as a valid email? Specifically, should we require for there to be at least two parts separated by a dot in the domain part?

In the tests, it is treated as a valid email.
However, in this issue, it seems to half-imply that we shouldn't treat it as a valid email.

**Comment by Richienb** [comment]
I just found your opinion in https://github.com/sindresorhus/email-regex/issues/2#issuecomment-506979632

Maybe we should add it as an option?

**Comment by sindresorhus** [comment]
Yeah, I think we should add an option and make it not accept domain-less emails by default.
